odxtools/parameters/tablekeyparameter.py

```diff
@@ -23,10 +23,16 @@
 class TableKeyParameter(Parameter):
 
     odx_id: OdxLinkId
+
+    # the spec mandates that exactly one of the two attributes must
+    # be non-None
     table_ref: Optional[OdxLinkRef]
     table_snref: Optional[str]
-    table_row_snref: Optional[str]
+
+    # the spec mandates that exactly one of the two attributes must
+    # be non-None
     table_row_ref: Optional[OdxLinkRef]
+    table_row_snref: Optional[str]
```

Comment:
New feature?

Reply:
not really. table rows can be referenced using a reference to a table plus a SNREF to a row in addition to a ODXLINK reference to the table row. the ODX XSD is pretty twisted here, so I probably missed this before...